test/integration-tests/V3ToV4Migration.test.ts

```diff
@@ -945,4 +961,301 @@ describe('V3 to V4 Migration Tests:', () => {
       })
     })
   })
+
+  describe('V4 Commands', () => {
+    beforeEach(async () => {
+      // initialize new pool on v4
+      await v4PositionManager.connect(bob).initializePool(
+        {
+          currency0: USDC.address,
+          currency1: WETH.address,
+          fee: FeeAmount.LOW,
+          tickSpacing: 10,
+          hooks: '0x0000000000000000000000000000000000000000',
+        },
+        '79228162514264337593543950336',
+        '0x'
+      )
+    })
+
+    it('mint v4 succeeds', async () => {
+      // transfer to v4posm
+      await usdcContract.connect(bob).transfer(v4PositionManager.address, expandTo6DecimalsBN(100000))
+      await wethContract.connect(bob).transfer(v4PositionManager.address, expandTo18DecimalsBN(100))
+
+      const encodedModifyLiquiditiesCallWithMint = mintSettleSweep()
+
+      planner.addCommand(CommandType.V4_POSITION_MANAGER_CALL, [encodedModifyLiquiditiesCallWithMint])
+
+      await executeRouter(planner, bob, router, wethContract, daiContract, usdcContract)
+
+      // bob successfully sweeped his usdc and weth from the v4 position manager
+      expect(await wethContract.balanceOf(v4PositionManager.address)).to.eq(0)
+      expect(await usdcContract.balanceOf(v4PositionManager.address)).to.eq(0)
+
+      // TODO: test that bob has the correct amount of weth and usdc
+    })
+
+    it('increase v4 succeeds', async () => {
+      // mint position first
+      await usdcContract.connect(bob).transfer(v4PositionManager.address, expandTo6DecimalsBN(100000))
+      await wethContract.connect(bob).transfer(v4PositionManager.address, expandTo18DecimalsBN(100))
+
+      const encodedModifyLiquiditiesCallWithMint = mintSettleSweep()
+
+      planner.addCommand(CommandType.V4_POSITION_MANAGER_CALL, [encodedModifyLiquiditiesCallWithMint])
+
+      await executeRouter(planner, bob, router, wethContract, daiContract, usdcContract)
+
+      // increase position second
+      planner = new RoutePlanner()
+      await usdcContract.connect(bob).transfer(v4PositionManager.address, expandTo6DecimalsBN(10000))
+      await wethContract.connect(bob).transfer(v4PositionManager.address, expandTo18DecimalsBN(10))
+
+      const encodedModifyLiquiditiesCallWithIncrease = increaseSettleSweep()
+
+      planner.addCommand(CommandType.V4_POSITION_MANAGER_CALL, [encodedModifyLiquiditiesCallWithIncrease])
+
+      await executeRouter(planner, bob, router, wethContract, daiContract, usdcContract)
+
+      // bob successfully sweeped his usdc and weth from the v4 position manager
+      expect(await wethContract.balanceOf(v4PositionManager.address)).to.eq(0)
+      expect(await usdcContract.balanceOf(v4PositionManager.address)).to.eq(0)
+    })
+
+    it('migrate with minting succeeds', async () => {
```

Comment:
i think maybe move migration tests into a new section?

so you have `V3 Commands` `V4 Commands` and then `Migration` or something?